src/pages/actions/useSideEffect.tsx

```diff
@@ -6,30 +6,45 @@ import {
   ReducerEvent,
   resetState
 } from "./events";
+import { LocalStorageLedgerActionStore } from "./ledgerActionStore";
 import { SideEffect } from "./reducer";
 
 export default function useSideEffect(
-  reload: () => void,
+  sideEffect: SideEffect | undefined,
   dispatch: React.Dispatch<ReducerEvent>,
-  sideEffect: SideEffect | undefined
+  reload: () => void
 ) {
   useEffect(() => {
     if (!sideEffect) {
       return;
     }
 
     switch (sideEffect.type) {
-      case "reloadData": {
-        reload();
-        return dispatch(resetState());
-      }
       case "executeAction": {
         executeAction(sideEffect.payload.action, sideEffect.payload.data).then(
-          response => dispatch(actionSuccessful(response)),
+          response =>
+            dispatch(
+              actionSuccessful(response, sideEffect.payload.action.name)
+            ),
           error => dispatch(actionFailed(error))
         );
         return;
       }
+      case "updateActionRecord": {
+        const ledgerActionStore = new LocalStorageLedgerActionStore(
+          window.localStorage
+        );
+        const actionName = sideEffect.payload.actionName;
+        const swapId = sideEffect.payload.swapId;
+        const transactionId = sideEffect.payload.transactionId;
+
+        ledgerActionStore.storeAction(actionName, swapId, transactionId);
+      }
+      // falls through
```

Comment:
On purpose? I'd rather like it so that the order of the switch statement is not important and everything that is necessary is contained within the `case` scope.

Reply:
On purpose! I hear you, but is it better to have to duplicate the reload logic? The linter forces you (via warning) to make this falling through explicit by adding the comment, so it's very hard to miss it.